apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -0,0 +1,245 @@
+import {
+  Media,
+  Project,
+  ProjectReport,
+  Site,
+  SiteReport,
+  Nursery,
+  NurseryReport
+} from "@terramatch-microservices/database/entities";
+import { MediaDto } from "../dto/media.dto";
+import { EntityModel, EntityType, EntityClass } from "@terramatch-microservices/database/constants/entities";
+import { AssociationProcessor } from "./association-processor";
+import { MediaQueryDto } from "../dto/media-query.dto";
+import { EntitiesService } from "../entities.service";
+import { DocumentBuilder, getDtoType, getStableRequestQuery } from "@terramatch-microservices/common/util";
+import { col, fn, Includeable, Op, Sequelize } from "sequelize";
+import { MediaAssociationDtoAdditionalProps } from "../dto/media-association.dto";
+import { Subquery } from "@terramatch-microservices/database/util/subquery.builder";
+import { Literal } from "sequelize/types/utils";
+
+type QueryModelType = {
+  modelType: string;
+  subquery: number[] | Literal;
+};
+
+export class MediaProcessor extends AssociationProcessor<Media, MediaDto> {
+  readonly DTO = MediaDto;
+
+  constructor(
+    protected readonly entityType: EntityType,
+    protected readonly entityUuid: string,
+    protected readonly entityModelClass: EntityClass<EntityModel>,
+    protected readonly entitiesService: EntitiesService
+  ) {
+    super(entityType, entityUuid, entityModelClass, entitiesService);
+  }
+
+  private async getSiteModels(site: Site): Promise<QueryModelType[]> {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [site.id] }];
+    const subquery = Subquery.select(SiteReport, "id").eq("siteId", site.id);
+    models.push({ modelType: SiteReport.LARAVEL_TYPE, subquery: subquery.literal });
+    return models;
+  }
+
+  private async getNurseryModels(nursery: Nursery) {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [nursery.id] }];
+    const subquery = Subquery.select(NurseryReport, "id").eq("nurseryId", nursery.id);
+    models.push({ modelType: NurseryReport.LARAVEL_TYPE, subquery: subquery.literal });
+    return models;
+  }
+
+  private async getProjectModels(project: Project) {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [project.id] }];
+
+    const projectReportSubquery = Subquery.select(ProjectReport, "id").eq("projectId", project.id);
+    models.push({ modelType: ProjectReport.LARAVEL_TYPE, subquery: projectReportSubquery.literal });
+
+    const siteSubquery = Subquery.select(Site, "id").eq("projectId", project.id);
+    models.push({ modelType: Site.LARAVEL_TYPE, subquery: siteSubquery.literal });
+
+    const sites = await Site.findAll({ where: { projectId: project.id }, attributes: ["id"] });
+
+    const siteReportSubquery = Subquery.select(SiteReport, "id").in(
+      "siteId",
+      sites.map(site => site.id)
```

Comment:
This should simply be `siteSubquery.literal`. The `Site.findAll()` should not be needed.

Reply:
changed